multiversx_sdk_rust_contract_builder/builder.py

```diff
@@ -92,12 +93,21 @@ def ensure_output_folder_is_empty(parent_output_folder: Path):
         raise ErrKnown(f"Output folder must be empty: {parent_output_folder}")
 
 
-def get_contracts_folders(project_path: Path) -> List[Path]:
+def get_contracts(project_path: Path):
```

Comment:
Perhaps keep this function as it was, and apply the check in a separate function `ensure_distinct_contract_names`.

Reply:
`ensure_distinct_contract_names` would duplicate the logic inside the for loop.

Would it be fine if check is done at the beginning of the for loop using a set `contract_names`?

Reply:
In this context, I think it's fine to have a loop and the calls to `get_contract_name_and_version` in `ensure_distinct_contract_names` (even if those exist in the main loop as well) - not duplication per-se.

Yes, using a set should also be all right.